zanata-war/src/main/java/org/zanata/service/impl/MergeTranslationsServiceImpl.java

```diff
@@ -231,6 +238,27 @@ private Integer mergeTranslations(
                         .getDocument().getId());
                 textFlowDAO.makePersistent(targetTf);
                 textFlowDAO.flush();
+
+                if (Events.exists() && !localeContentStateMap.isEmpty()) {
+                    for (Map.Entry<Long, ContentState> entry : localeContentStateMap
+                            .entrySet()) {
+                        HTextFlowTarget updatedTarget =
+                                targetTf.getTargets().get(entry.getKey());
+
+                        TextFlowTargetStateEvent event =
+                                new TextFlowTargetStateEvent(null,
+                                        targetVersionId,
+                                        targetTf.getDocument().getId(),
+                                        targetTf.getId(),
+                                        updatedTarget.getLocale().getLocaleId(),
+                                        updatedTarget.getId(),
+                                        updatedTarget.getState(),
+                                        entry.getValue());
+
+                        Events.instance().raiseTransactionSuccessEvent(
+                                TextFlowTargetStateEvent.EVENT_NAME, event);
```

Comment:
That will be more change - new Event type to take in list of informations which I tried to minimise.